common/str.cpp

```diff
@@ -292,6 +293,13 @@ String &String::operator+=(const char *str) {
 	return *this;
 }
 
+bool String::pointerInOwnBuffer(const char *str) const {
+	uintptr_t ownBuffStart = (uintptr_t)_str;
```

Comment:
It is not allowed to use standard types outside of `backends/` dure to portability reasons.